setup/steps/configure_log_location.go

```diff
@@ -0,0 +1,163 @@
+package steps
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	survey "github.com/AlecAivazis/survey/v2"
+	"github.com/go-ini/ini"
+	"github.com/pganalyze/collector/setup/query"
+	"github.com/pganalyze/collector/setup/state"
+	s "github.com/pganalyze/collector/setup/state"
+)
+
+var ConfigureLogLocation = &s.Step{
+	Kind:        state.LogInsightsStep,
+	Description: "Determine log location",
+	Check: func(state *s.SetupState) (bool, error) {
+		return state.CurrentSection.HasKey("db_log_location"), nil
+	},
+	Run: func(state *s.SetupState) error {
+		guessedLogLocation, err := discoverLogLocation(state.CurrentSection, state.QueryRunner)
+		if err != nil {
+			return err
+		}
+		var logLocationConfirmed bool
+		if state.Inputs.Scripted {
+			if state.Inputs.GuessLogLocation.Valid && state.Inputs.GuessLogLocation.Bool {
+				if state.Inputs.Settings.DBLogLocation.Valid && state.Inputs.Settings.DBLogLocation.String != "" {
+					return errors.New("cannot specify both guess_log_location and set explicit db_log_location")
+				}
+				logLocationConfirmed = state.Inputs.GuessLogLocation.Bool
+			}
+		} else {
+			err = survey.AskOne(&survey.Confirm{
+				Message: fmt.Sprintf("Your database log file or directory appears to be %s; is this correct (will be saved to collector config)?", guessedLogLocation),
+				Default: false,
+			}, &logLocationConfirmed)
+			if err != nil {
+				return err
+			}
+		}
+
+		var logLocation string
+		if logLocationConfirmed {
+			logLocation = guessedLogLocation
+		} else if state.Inputs.Scripted {
+			if !state.Inputs.Settings.DBLogLocation.Valid || state.Inputs.Settings.DBLogLocation.String == "" {
+				return errors.New("db_log_location not provided and guess_log_location flag not set")
+			}
+			logLocation = state.Inputs.Settings.DBLogLocation.String
+		} else {
+			err = survey.AskOne(&survey.Input{
+				Message: "Please enter the Postgres log file location (will be saved to collector config)",
+			}, &logLocation, survey.WithValidator(validatePath))
+		}
+		_, err = state.CurrentSection.NewKey("db_log_location", logLocation)
+		if err != nil {
+			return err
+		}
+		return state.SaveConfig()
+	},
+}
+
+func validatePath(ans interface{}) error {
+	ansStr, ok := ans.(string)
+	if !ok {
+		return errors.New("expected string value")
+	}
+	// TODO: also confirm this is readable by the regular pganalyze user
```

Comment:
For now I think its fine to not check permissions here - in the future we should have a follow-up step that checks permissions (but we can do this as follow-up work, since as discussed on Ubuntu-based systems the logs should be accessible for the `adm` group)